subsys/nrf_security/src/drivers/cracen/cracenpsa/src/cracen_lm20_mac_cmac.c

```diff
@@ -0,0 +1,217 @@
+#include <stdint.h>
+#include <string.h>
+#include <psa/crypto.h>
+#include <sxsymcrypt/blkcipher.h>
+#include <sxsymcrypt/keyref.h>
+#include "cracen_lm20_mac_cmac.h"
+#include <sxsymcrypt/aes.h>
+
+#define AES_BLOCK_SIZE	  (16)
+#define CMAC_PADDING_BYTE (0x80)
+#define AES_CMAC_MSB	  (0x80)
+#define CMAC_CONSTANT_RB  (0x87)
+
+static psa_status_t cracen_aes_ecb_encrypt(const struct sxkeyref *key, const uint8_t *input,
+					   uint8_t *output)
+{
+	struct sxblkcipher blkciph;
+	int sx_status;
+
+	sx_status = sx_blkcipher_create_aesecb_enc(&blkciph, key);
+	if (sx_status != SX_OK) {
+		return silex_statuscodes_to_psa(sx_status);
+	}
+
+	sx_status = sx_blkcipher_crypt(&blkciph, input, AES_BLOCK_SIZE, output);
+	if (sx_status != SX_OK) {
+		return silex_statuscodes_to_psa(sx_status);
+	}
+
+	sx_status = sx_blkcipher_run(&blkciph);
+	if (sx_status != SX_OK) {
+		return silex_statuscodes_to_psa(sx_status);
+	}
+
+	sx_status = sx_blkcipher_wait(&blkciph);
+	if (sx_status != SX_OK) {
+		return silex_statuscodes_to_psa(sx_status);
+	}
+
+	return PSA_SUCCESS;
+}
+
+psa_status_t cracen_lm20_cmac_setup(cracen_mac_operation_t *operation,
+				    const psa_key_attributes_t *attributes,
+				    const uint8_t *key_buffer, size_t key_buffer_size)
+{
+	psa_status_t status = PSA_SUCCESS;
+	int sx_status;
+
+	/* Only AES-CMAC is supported */
+	if (psa_get_key_type(attributes) != PSA_KEY_TYPE_AES) {
+		return PSA_ERROR_NOT_SUPPORTED;
+	}
+
+	/* lm20 does not support 192-bit keys */
```

Comment:
Ignore this comment. Apparently we do and no one documented it. Will remove after ci